src/core/credentials/transport/tls/spiffe_utils.cc

```diff
@@ -111,19 +120,16 @@ absl::Status validatePath(absl::string_view path) {
 
 }  // namespace
 
-// spiffe://trust-domain-name/path
 absl::StatusOr<SpiffeId> SpiffeId::FromString(absl::string_view uri) {
-  if (absl::Status status = doInitialUriValidation(uri); !status.ok()) {
-    return status;
-  }
-  if (!absl::StartsWithIgnoreCase(uri, SPIFFE_PREFIX)) {
+  GRPC_RETURN_IF_ERROR(DoInitialUriValidation(uri));
+  if (!absl::StartsWithIgnoreCase(uri, kSpiffePrefix)) {
     return absl::InvalidArgumentError("SPIFFE ID must start with spiffe://");
   }
   if (absl::EndsWith(uri, /*suffix=*/"/")) {
     return absl::InvalidArgumentError("SPIFFE ID cannot end with a /");
   }
   // The input definitely starts with spiffe://
-  absl::string_view trust_domain_and_path = uri.substr(SPIFFE_PREFIX.length());
+  absl::string_view trust_domain_and_path = uri.substr(kSpiffePrefix.length());
```

Comment:
You can probably avoid this by using `absl::ConsumePrefix()` instead of `absl::StartsWith()` above.